### ISSUE TITLE
Debugger: `cycles` seems to be useless

### ISSUE THREAD

**Issue by Stevie-O** [mamedev/mame#6654]
While trying to analyze a game's behavior, I wanted to get some timing information.

There's a field displayed in the main debugger window called "cycles", and there's an associated symbol "cycles".  This value was way too small to be correct.   I had to dive into the source code to find out that this symbol means "number of cycles left in the current timeslice."

Is this information actually useful to anyone?

I propose that:

* `cycles` be renamed to `cyclesleft`
* `totalcycles` be renamed to either `cycle` or `cycles` (an alias could be added to proxy `totalcycles`)
    Rationale for `cycle`: the number of frames completed so far is named `frame`, not `frames`, also less ambiguity between versions
* The value displayed in the debugger window should be the one corresponding to (original) `totalcycles`, not (original) `cycles`.  Or, if someone really wants (new) `cyclesleft`, display both.  There seems to be plenty of room.

**Comment by MooglyGuy** [comment]
I agree with your assessment and your proposed fixes, but fair warning, the MAME team tends to react negatively to people bossing them around.

My advice would be to implement these changes yourself and submit a pull request. I'll be only too happy to merge it in and deal with whatever hail of verbal bullets comes my way.

**Comment by Stevie-O** [comment]
Oh, is absolutely my intention to implement this myself.  (I have a list of about 40 separate issues I've run into during my first experience involving the MAME debugger.  I plan to either do them myself or write off the idea as infeasible.) 

Most of these things are for obvious bugs or completely new functionality that I don't expect anyone to have objections to.  I'm gonna implement them, test them, and if they work out the way I expect them to, just create a PR for them.

Others, however, such as this one, the `symlist` one (#6655), and a few others I'm putting together, I wanted some feedback before I expended the effort to make the changes.  The main purpose of these tickets is to get feedback on my proposals before I write something that people aren't happy with.

**Comment by rb6502** [comment]
Cycles left in the current timeslice is vital to debugging MAME and its drivers.  I can see how it's not interesting if you're debugging software running on the emulated machine though.  Please do not get rid of it.

**Comment by MooglyGuy** [comment]
I do have to concur with @rb6502 that the value is useful, but as a reverse-engineering and documentation tool, I would personally prefer that the executed number of cycles be front and center, with the cycles left shown on a line immediately below it.

Also, I have to say, that's a remarkably level-headed and productive way of approaching things.

Usually the team has to brook people complaining that their 2005-era ROM sets no longer work, or that they're not getting good performance on an absolute madman of a performance-soak driver.

If you've thought it out this far, then all I can say is go for it - you'll probably get feedback that involves minor tweaks to your plan, but you obviously know what you're doing, so by all means have a stab at it. The MAME debugger has been in need of some love for a while now.

**Comment by ajrhacker** [comment]
> Cycles left in the current timeslice is vital to debugging MAME and its drivers. I can see how it's not interesting if you're debugging software running on the emulated machine though.

The point at which that number becomes uninteresting, in my experience, is when the timeslices are cut so thinly that it never rises above 1. (This commonly happens with a MCS-48 microcontroller taking 15 or 30 clocks for each cycle while a likely-unrelated UART is being fed with a periodic timer firing at 32 times the baud rate.)

I agree that the number of total cycles would be useful to see.

**Comment by Stevie-O** [comment]
@rb6502 Okay, I had suspected that it might have been there for that purpose (i.e. development of MAME itself).

Do you have any particular issue with me renaming `cycles` to `cyclesleft`? (I'll ensure that accessing it as `cycles` continues to work from debugger commands -- the situation you describe will almost certainly doing the same thing on two different MAME versions and seeing what differs, so requiring different commands across versions would be very inconvenient.)  But in the UI, I'll label it `cyclesleft` instead of `cycles`.

**Comment by rb6502** [comment]
I don't have any particular problem.  Lemme add in the Big Guns though, @cuavas.
